qiita_db/util.py

```diff
@@ -1665,23 +1670,37 @@ def get_artifacts_information(artifact_ids, only_biom=True):
                             qdb.sql_connection.TRN.execute_fetchflatten()
                     target = ts[target]
 
-                if prep_template_id is None:
-                    prep_samples = 0
-                else:
+                prep_samples = 0
+                platform = ''
+                target_gene = ''
+                if prep_template_id is not None:
                     if prep_template_id not in ps:
-                        ps[prep_template_id] = len(list(
-                            PT(prep_template_id).keys()))
-                    prep_samples = ps[prep_template_id]
+                        pt = PT(prep_template_id)
+                        categories = pt.categories()
+                        if 'platform' in categories:
```

Comment:
I defer on this to @wasade, but I think instead of an empty string for platform and target_gene, it would likely be better to return "Not Available" or something along those lines.

Reply:
Not sure, i thought it was "not available" lowercase. Best to ask Gail

Reply:
EBI is "null values: not applicable, missing: not collected, missing: not provided, missing: restricted access.". However, this is different and the current use case is redbiom, so whatever you prefer @wasade .

Reply:
After discussing offline with @wasade we are going with: 'not provided'